goods/views.py

```
@@ -218,7 +250,62 @@ def post(self, request, pk):
         return JsonResponse(data={"good": good_data}, status=status.HTTP_200_OK)
 
 
-class GoodDetail(APIView):
+class GoodDetails(APIView):
```

If you use a retrieve update api view, you can just raise the errors below instead of creating whole jsonresponses, and it will handle the response etc for you. Would make the code a bit easier to read, even if you just writing your own get/put.

---

Nice, thanks